### PR TITLE
Moved CompactionCoordinator into Manager

### LINKED ISSUE #3214: Offline Operations: Move CompactionCoordinator into Manager
Move the CompactionCoordinator logic into the Manager. The Compactor's will need to be modified to contact the Manager for job assignments.

### PR DESCRIPTION
**Pull request by dlmarion**
Closes #3214

### CODE REVIEW

**Comment by dlmarion** [comment]
Full IT build succeeded

**Review by keith-turner — APPROVED** [review]
Ran some local greps on this on the branch and found the following that seems like it could be removed.

```
./assemble/conf/accumulo-env.sh:  compaction-coordinator) JAVA_OPTS=('-Xmx512m' '-Xms512m' "${JAVA_OPTS[@]}") ;;
./assemble/conf/accumulo-env.sh:  monitor | gc | manager | tserver | compaction-coordinator | compactor | sserver)
```

File: server/manager/src/main/java/org/apache/accumulo/manager/compaction/DeadCompactionDetector.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ private void detectDeadCompactions() {
 
     if (tabletCompactions.isEmpty()) {
       // Clear out dead compactions, tservers don't think anything is running
-      log.trace("Clearing the dead compaction map, no tablets have compactions running");
+      log.debug("Clearing the dead compaction map, no tablets have compactions running");
```
[the remark this answers is not in the document]
Comment (dlmarion) [reply server/manager/src/main/java/org/apache/accumulo/manager/compaction/DeadCompactionDetector.java]: I don't remember the reason. Reverted in bbf2fe2

File: server/base/src/main/java/org/apache/accumulo/server/init/ZooKeeperInitializer.java [hunk 2]
```diff
@@ -156,10 +156,6 @@ void initialize(final ServerContext context, final boolean clearInstanceName,
         ZooUtil.NodeExistsPolicy.FAIL);
     zoo.putPersistentData(zkInstanceRoot + WalStateManager.ZWALS, EMPTY_BYTE_ARRAY,
         ZooUtil.NodeExistsPolicy.FAIL);
-    zoo.putPersistentData(zkInstanceRoot + Constants.ZCOORDINATOR, EMPTY_BYTE_ARRAY,
```
[the remark this answers is not in the document]
Comment (dlmarion) [reply server/base/src/main/java/org/apache/accumulo/server/init/ZooKeeperInitializer.java:159]: Good catch, removed constants in bbf2fe2.

File: core/src/main/java/org/apache/accumulo/core/conf/cluster/ClusterConfigParser.java [hunk 3]
```diff
@@ -95,10 +95,6 @@ public static void outputShellVariables(Map<String,String> config, PrintStream o
       }
     }
 
-    if (config.containsKey("compaction.coordinator")) {
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/conf/cluster/ClusterConfigParser.java:98]: #3452 is merged into elasticity, so can remove compaction coordinator from that new validation

[the remark this answers is not in the document]
Comment (dlmarion) [reply core/src/main/java/org/apache/accumulo/core/conf/cluster/ClusterConfigParser.java:115]: Removed in ad4e22e

**Pull request merged**